tests/Doctrine/Migrations/Tests/DependencyFactoryTest.php

```diff
@@ -156,6 +158,36 @@ public function testServiceDefinition() : void
         self::assertSame($anotherLogger, $di->getLogger());
     }
 
+    public function testServiceDefinitionWithReference() : void
+    {
+        $migration = $this->createMock(AbstractMigration::class);
+
+        $di = DependencyFactory::fromConnection(new ExistingConfiguration($this->configuration), new ExistingConnection($this->connection));
+
+        $di->setDefinition(MigrationFactory::class, static function (DependencyFactory $innerDi) use ($migration) {
+            // this call needs to stay here just to trigger again the dependency resolution
```

Comment:
why do we need this exactly ? This creates a confusing API IMO

Reply:
This is just because I was lazy to create some service that effectively decorates the one returned by `getMigrationFactory()`. Will create one and use that argument.